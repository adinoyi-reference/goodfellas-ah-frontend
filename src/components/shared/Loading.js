import React from 'react';
import { spinner } from '../../mixin';

export const Loader = () => (
  <div className="no-record centralizer">
    <div><img src={spinner} alt="loader" />&nbsp;&nbsp;
      <span>Loading</span>
    </div>
  </div>
);

const Loading = () => (
  <div className="loading-icon">
    <img src={spinner} alt="" />
  </div>
);

export default Loading;
