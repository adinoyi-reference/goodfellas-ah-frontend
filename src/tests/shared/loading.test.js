import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../../components/shared/Loading';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <Loading />
    </Root>,
  );
});

afterEach(() => wrapped.unmount());
describe('Loading', () => {
  describe('render', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
