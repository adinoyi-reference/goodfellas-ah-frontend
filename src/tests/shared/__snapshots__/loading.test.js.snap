// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading render container should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Loading>
      <div
        className="loading-icon"
      >
        <img
          alt=""
          src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541426069/Authors%20Haven/Spinner-1s.gif"
        />
      </div>
    </Loading>
  </Provider>
</_default>
`;
